File: xsolla-store-sdk/src/main/java/com/xsolla/android/store/entity/response/common/IPrice.java
package com.xsolla.android.store.entity.response.common;

import java.math.BigDecimal;

public interface IPrice {

    String getCurrencyId();

    String getCurrencyName();

    String getAmountRaw();

    String getAmountWithoutDiscountRaw();

    BigDecimal getAmountDecimal();

    BigDecimal getAmountWithoutDiscountDecimal();

}

File: xsolla-store-sdk/src/main/java/com/xsolla/android/store/entity/response/common/IPrice.kt
package com.xsolla.android.store.entity.response.common

import java.math.BigDecimal

interface IPrice {
    fun getCurrencyId(): String?

    fun getCurrencyName(): String?

    fun getAmountRaw(): String?

    fun getAmountWithoutDiscountRaw(): String?

    fun getAmountDecimal(): BigDecimal?

    fun getAmountWithoutDiscountDecimal(): BigDecimal?
}